src/shared/common/forms/composite/somu-list.jsx

```diff
@@ -22,112 +24,70 @@ const STATUS_OPTION = {
 
 class SomuTableRenderer {
 
-    constructor(renderer, choices) {
+    constructor(renderer, choices ) {
         this.state = { renderer, choices };
     }
 
-    parseDate(rawDate){
-        const [date] = rawDate.match(/\d{4}-(0[1-9]|1[012])-(0[1-9]|[12][0-9]|3[01])/g) || [];
-        if (!date) {
-            return null;
-        }
-        const [year, month, day] = date.split('-');
-        return `${day}/${month}/${year}`;
-    }
-
-    formatDate(date) {
-        return date ? this.parseDate(date) : null;
-    }
-
-    loadValue(value, choices) {
-        const choice = choices.find(x => x.value === value);
-        return choice ? choice.label : value;
-    }
-
-    addDays(dateString, days) {
-        const date = new Date(dateString);
-        date.setDate(date.getDate() + days);
-        return date;
-    }
-
     getRequestStatus(dueDate, status) {
         if (STATUS_OPTION.CANCELLED.includes(status)) {
             return REQUEST_STATUS.CANCELLED;
         } else if (STATUS_OPTION.COMPLETE.includes(status)) {
             return REQUEST_STATUS.COMPLETE;
-        } else if (this.addDays(dueDate, 1) < Date.now()) {
+        } else if (addDaysToDate(dueDate, 1) < Date.now()) {
             return REQUEST_STATUS.OVERDUE;
         } else {
             return REQUEST_STATUS.DUE;
         }
     }
 
-    composeTitle(choices, titleBusinessLabelData) {
-        const { businessArea, businessUnit } = titleBusinessLabelData;
-
-        if (businessArea && businessUnit) {
-            return `${businessArea} - ${this.loadValue(businessUnit, choices)}`;
+    composeContributionTitle(contributionBusinessArea, contributionBusinessUnit) {
+        if (contributionBusinessArea && contributionBusinessUnit) {
+            return `${this.getChoicesValue({ contributionBusinessArea })} - ${this.getChoicesValue({ contributionBusinessUnit }) }`;
         }
-
-        if (businessArea && !businessUnit) {
-            return `${businessArea}`;
+        if (contributionBusinessUnit) {
+            return this.getChoicesValue({ contributionBusinessUnit });
         }
-        return `${this.loadValue(businessUnit, choices)}`;
+
+        return this.getChoicesValue({ contributionBusinessArea });
     }
 
-    renderContributionTable(somuItem) {
+    getChoicesValue(choice) {
         const { choices } = this.state;
+        const [name, value = ''] = getObjectNameValue(choice);
+
+        const choiceList = choices[name] ?? [];
+        return choiceList.find(item => item.value === value)?.label ?? value;
+    }
```

Comment:
So this is the section that allows you to get the full title for the contribution?

Reply:
Yes - we pass in an object with key of the field and value of an array of choices.